### DISCUSSION TITLE
Ice 0.10.0

### DISCUSSION THREAD

**Discussion by jordanbaird** [jordanbaird/Ice#218] [Announcements]
## What's New
* Menu bar items can now be shown in a secondary bar below the menu bar! (look for "Use Ice Bar" in the "General" settings pane)
* Added ability to move menu bar items individually in a new "Menu Bar Items" settings pane
* Menu bar appearance can now be inset on screens with a notch
* Ice now ensures that the always-hidden section stays to the left of the hidden section
* Added the ability to disable the always-hidden section
* Ice now launches in the background by default
* Ice now uses the GPL-3.0 license

## Bug Fixes
* The menu bar appearance overlay should now work with fullscreen in apps that don't use the default macOS fullscreen interface, such as VLC and Keynote
* Reliably check for default macOS fullscreen spaces
* Fixed mouse location checks in fullscreen and when using multiple displays
* The menu bar appearance now renders at the correct size across multiple displays with different sized menu bars (i.e. when one display has a notch and the other does not)
* Fixed screen flicker when playing DRM media
* Improved reliability of application menu hiding
* Fixed excessive CPU usage by menu bar overlay panel when switching apps

## New Contributors
* @smbl64 made their first contribution in https://github.com/jordanbaird/Ice/pull/126
* @eltociear made their first contribution in https://github.com/jordanbaird/Ice/pull/132

**Full Changelog**: https://github.com/jordanbaird/Ice/compare/0.9.0...0.10.0

<hr /><em>This discussion was created from the release <a href='https://github.com/jordanbaird/Ice/releases/tag/0.10.0'>Ice 0.10.0</a>.</em>

**Comment by Fogh** [comment]
I just updated, but it seems that the app is broken. The hidden section is always shown and clicking the ice icon does nothing. 

![Screen Recording 2024-07-09 at 10 10 03](https://github.com/jordanbaird/Ice/assets/434624/8565b8f5-fbd9-4d39-96b7-c519ffc78811)

**Reply by jordanbaird** [reply]
Where do the section dividers appear when you hold Command and drag an icon? There should be a big and small arrow.

**Reply by CyberHaik** [reply]
For me, the visible section in menu bar items is gone. Only hidden and always hidden remain.

**Reply by Fogh** [reply]
> Where do the section dividers appear when you hold Command and drag an icon? There should be a big and small arrow.

Only the always hidden shows. Same when I alt click the Ice icon.  The always hidden section works as intended it seems.

**Reply by jordanbaird** [reply]
@Fogh I think I just realized what's going on. To verify, could you open Terminal and run the following command:

```
defaults read com.jordanbaird.Ice
```

And then paste the output here?

**Reply by jordanbaird** [reply]
@hisikbay Is it completely gone? Or is the section there, but just empty?

**Reply by Fogh** [reply]
@jordanbaird 
```plist
{
    AboutSettingsPaneSelectedTab = 1;
    AutoRehide = 1;
    CanToggleAlwaysHiddenSection = 1;
    CustomIceIconIsTemplate = 0;
    EnableAlwaysHiddenSection = 1;
    HideApplicationMenus = 1;
    Hotkeys =     {
        ShowSectionDividers = {length = 4, bytes = 0x6e756c6c};
        ToggleAlwaysHiddenSection = {length = 4, bytes = 0x6e756c6c};
        ToggleApplicationMenus = {length = 4, bytes = 0x6e756c6c};
        ToggleHiddenSection = {length = 4, bytes = 0x6e756c6c};
    };
    IceBarLocation = 0;
    IceIcon = {length = 93, bytes = 0x7b226869 6464656e 223a7b22 63617461 ... 726f6b65 227d7d7d };
    MenuBarAppearanceConfiguration = {length = 21062, bytes = 0x7b226861 73536861 646f7722 3a66616c ... 3a5b302c 315d7d7d };
    MenuBarSettingsPaneSelectedTab = 0;
    "NSSplitView Subview Frames NSColorPanelSplitView" =     (
        "0.000000, 0.000000, 224.000000, 258.000000, NO, NO",
        "0.000000, 259.000000, 224.000000, 48.000000, NO, NO"
    );
    "NSSplitView Subview Frames SettingsWindow, SidebarNavigationSplitView" =     (
        "0.000000, 0.000000, 210.000000, 600.000000, NO, NO",
        "211.000000, 0.000000, 689.000000, 600.000000, NO, NO"
    );
    "NSStatusItem Preferred Position AHItem" = 781;
    "NSStatusItem Preferred Position SItem" = 383;
    "NSStatusItem Visible AHItem" = 1;
    "NSStatusItem Visible HItem" = 0;
    "NSStatusItem Visible SItem" = 1;
    "NSToolbar Configuration com.apple.NSColorPanel" =     {
        "TB Is Shown" = 1;
    };
    "NSWindow Frame NSColorPanel" = "0 97 224 278 0 0 2560 1415 ";
    "NSWindow Frame PermissionsWindow" = "1078 519 403 445 0 0 2560 1415 ";
    "NSWindow Frame SettingsWindow" = "830 632 900 600 0 0 2560 1415 ";
    RehideInterval = 15;
    RehideStrategy = 0;
    SUAutomaticallyUpdate = 1;
    SUEnableAutomaticChecks = 1;
    SUHasLaunchedBefore = 1;
    SULastCheckTime = "2024-07-09 08:12:09 +0000";
    SUSendProfileInfo = 0;
    SUUpdateGroupIdentifier = 142590880;
    ShowIceIcon = 1;
    ShowOnClick = 1;
    ShowOnHover = 0;
    ShowOnScroll = 1;
    ShowSectionDividers = 1;
    UseIceBar = 0;
    "hasMigrated0_10_0" = 1;
    "hasMigrated0_8_0" = 1;
}
```

**Reply by jordanbaird** [reply]
@Fogh Yep. For the the time being, you can fix the problem by running:

```
defaults write com.jordanbaird.Ice "NSStatusItem Visible AHItem" -bool true;
defaults write com.jordanbaird.Ice "NSStatusItem Visible HItem" -bool true;
defaults write com.jordanbaird.Ice "NSStatusItem Visible SItem" -bool true 
```

Then, restart Ice.

I'll get a hotfix out as soon as possible.

- Note: For others that are having this same issue, you may have to move whichever arrow was hidden back to its original location after running this command, as it may have been inserted at the beginning of the menu bar. You can hold Command and drag any item to show the arrows.

**Reply by Fogh** [reply]
@jordanbaird Thanks, that fixed the problem
